src/shared/components/palette/complex/read-complex-field-collection-table.component.spec.ts

```diff
@@ -0,0 +1,213 @@
+          imports: [
+            PaletteUtilsModule,
+            ConditionalShowModule
+          ],
+          declarations: [
+            ReadComplexFieldCollectionTableComponent,
+            FieldsFilterPipe,
+
+            // Mock
+            FieldReadComponent,
+          ],
+          providers: []
+        })
+        .compileComponents();
+
+      fixture = TestBed.createComponent(ReadComplexFieldCollectionTableComponent);
+      component = fixture.componentInstance;
+
+      component.caseField = CASE_FIELD;
+      component.context = PaletteContext.TABLE_VIEW;
+
+      de = fixture.debugElement;
+      fixture.detectChanges();
+    }));
+
+    it('should render a panel with a header for the complex type', () => {
+      let panelTitle = de
+        .query($COMPLEX_PANEL)
+        .query($COMPLEX_PANEL_TITLE);
+
+      expect(panelTitle).toBeTruthy();
+      expect(panelTitle.nativeElement.textContent).toBe(CASE_FIELD.label);
+    });
+
+    it('should render a table with a row containing 2 columns for each simple type', () => {
+      let simpleRowsHeaders = de
+        .query($COMPLEX_PANEL)
+        .queryAll($COMPLEX_PANEL_SIMPLE_ROWS_HEADERS);
+      expect(simpleRowsHeaders.length).toBe(2);
+      expect(simpleRowsHeaders[LINE_1].nativeElement.textContent).toBe(FIELD_TYPE_WITH_VALUES.complex_fields[LINE_1].label);
+      expect(simpleRowsHeaders[LINE_2].nativeElement.textContent).toBe(FIELD_TYPE_WITH_VALUES.complex_fields[LINE_2].label);
+
+      let simpleRowsValues = de
+        .query($COMPLEX_PANEL)
+        .queryAll($COMPLEX_PANEL_SIMPLE_ROWS_VALUES);
+      expect(simpleRowsValues.length).toBe(4);
+      expect(simpleRowsValues[0].componentInstance.caseField.value).toEqual(CASE_FIELD.value[0].value[FIRST_COLUMN]);
+      expect(simpleRowsValues[1].componentInstance.caseField.value).toEqual(CASE_FIELD.value[0].value[SECOND_COLUMN]);
+      expect(simpleRowsValues[2].componentInstance.caseField.value).toEqual(CASE_FIELD.value[1].value[FIRST_COLUMN]);
+      expect(simpleRowsValues[3].componentInstance.caseField.value).toEqual(CASE_FIELD.value[1].value[SECOND_COLUMN]);
+
+      let simpleRowsHeadersClickers = de
+        .query($COMPLEX_PANEL)
+        .queryAll($COMPLEX_PANEL_SIMPLE_ROWS_HEADERS_CLICK);
+      expect(simpleRowsHeadersClickers.length).toBe(2);
+
+      expect(simpleRowsHeadersClickers[0].properties.innerHTML).toEqual(UNORDERED);
+      expect(simpleRowsHeadersClickers[1].properties.innerHTML).toEqual(UNORDERED);
+
+    });
+
+    it('should sort rows based on column name', () => {
+      expect(component.rows[0].AddressLine1).toEqual(CASE_FIELD.value[0].value[FIRST_COLUMN]);
+      component.sortRowsByColumns('AddressLine1');
+      expect(component.rows[0].AddressLine1).toEqual(CASE_FIELD.value[1].value[FIRST_COLUMN]);
+      component.sortRowsByColumns('AddressLine1');
+      expect(component.rows[0].AddressLine1).toEqual(CASE_FIELD.value[0].value[FIRST_COLUMN]);
+
+    });
+
```

Comment:
Can we have few more tests that verify sorting of other types then just `Text`?

Reply:
Added another test for Number that should also do complete code coverage